pkg/generic/httpthrift_codec.go

```diff
@@ -46,15 +46,37 @@ type HTTPRequest = descriptor.HTTPRequest
 type HTTPResponse = descriptor.HTTPResponse
 
 type httpThriftCodec struct {
-	svcDsc           atomic.Value // *idl
-	provider         DescriptorProvider
-	codec            remote.PayloadCodec
-	binaryWithBase64 bool
+	svcDsc              atomic.Value // *idl
+	provider            DescriptorProvider
+	codec               remote.PayloadCodec
+	binaryWithBase64    bool
+	enableDynamicgoReq  bool
+	enableDynamicgoResp bool
+	convOpts            conv.Options
 }
 
-func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec) (*httpThriftCodec, error) {
+func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec, opts ...Options) (*httpThriftCodec, error) {
 	svc := <-p.Provide()
-	c := &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	var c *httpThriftCodec
+	if opts == nil {
+		c = &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	} else {
+		// codec with dynamicgo
+		convOpts := opts[0].DynamicgoConvOpts
```

Comment:
I think the logic here is a little confusing, maybe you can provide a default options config, and let users to pass their's customized config.

Reply:
fixed